### PR TITLE
LG-2601: consume darkMode from LG provider

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-2601](https://jira.mongodb.org/browse/LG-2601)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

**Comment by spark33** [comment]
> Can we add tests to each of these components that explicitly proves we are able to both read darkmode from the provider (both when darkmode is true and when darkmode is false) and also that we're able to successfully override the value from the component?

Could we define it in `internal` so we can import the test suite in every component? It'll be a pretty broadly used pattern.

**Comment by shaneeza** [comment]
@brookescarlett @spark33 What is `internal`?

**Comment by brookescarlett** [comment]
> @brookescarlett @spark33 What is `internal`?

Would actually be from `lib`

File: packages/combobox/src/Chip.tsx [hunk 1]
```diff
@@ -3,10 +3,10 @@ import Icon from '@leafygreen-ui/icon';
 import { css, cx } from '@leafygreen-ui/emotion';
 import { palette } from '@leafygreen-ui/palette';
 import InlineDefinition from '@leafygreen-ui/inline-definition';
-import { keyMap } from '@leafygreen-ui/lib';
+import { keyMap, Theme } from '@leafygreen-ui/lib';
 import { transitionDuration, typeScales } from '@leafygreen-ui/tokens';
-import { ChipProps, ComboboxSize as Size, Theme } from './Combobox.types';
-import { ComboboxContext, useDarkMode } from './ComboboxContext';
+import { ChipProps, ComboboxSize as Size } from './Combobox.types';
```
Comment (brookescarlett) [inline packages/combobox/src/Chip.tsx:8]: can we rm theme from Combobox types if it lives in lib now?

File: packages/combobox/src/Combobox.tsx [hunk 2]
```diff
@@ -44,7 +45,7 @@ import {
   getValueForDisplayName,
   getNameAndValue,
 } from './utils';
-import { ComboboxContext, useDarkMode } from './ComboboxContext';
+import { ComboboxContext } from './ComboboxContext';
```
Comment (brookescarlett) [inline packages/combobox/src/Combobox.tsx:48]: ditto the local useDarkMode

File: packages/combobox/src/Combobox.types.ts [hunk 3]
```diff
@@ -1,13 +1,6 @@
 import { ReactElement, ReactNode } from 'react';
 import { Either, HTMLElementProps } from '@leafygreen-ui/lib';
 
-export const Theme = {
```
Comment (brookescarlett) [inline packages/combobox/src/Combobox.types.ts:4]: beat me to it!

File: packages/combobox/src/Combobox.tsx (hunk 2 above)
Comment (shaneeza) [reply packages/combobox/src/Combobox.tsx:48]: this one was remove already

File: packages/combobox/src/ComboboxContext.tsx [hunk 4]
```diff
@@ -24,13 +25,11 @@ interface ComboboxData {
...
+  theme: Theme.Light,
   size: ComboboxSize.Default,
   withIcons: false,
   disabled: false,
   isOpen: false,
   state: State.none,
   searchState: SearchState.unset,
 });
-
-export const useDarkMode = (darkMode: boolean) =>
```
Comment (shaneeza) [inline packages/combobox/src/ComboboxContext.tsx:35]: @brookescarlett here

**Review by brookescarlett — APPROVED** [review]
